lib/rucio/daemons/storage/consistency/actions.py

```diff
@@ -0,0 +1,805 @@
+
+    return unknown_replicas
+### NOTE: declare_bad_file_replicas will be used directly from core/replica.py
+### when handling of DID is added there
+##########################################################################
+
+
+##############################
+### This is Igor's Stats class
+class Stats(object):
+
+    def __init__(self, path):
+        self.Path = path
+        self.Data = {}
+
+    def __getitem__(self, name):
+        return self.Data[name]
+
+    def __setitem__(self, name, value):
+        self.Data[name] = value
+        self.save()
+
+    def get(self, name, default=None):
+        return self.Data.get(name, default)
+
+    def update(self, data):
+        self.Data.update(data)
+        self.save()
+
+    def save(self):
+        try:
+            with open(self.Path, "r") as f:
+                data = f.read()
+        except:
+            #traceback.print_exc()
+            data = ""
+        data = json.loads(data or "{}")
+        data.update(self.Data)
+        open(self.Path, "w").write(json.dumps(data, indent=4))
+
+
+def write_stats(my_stats, stats_file, stats_key=None):
+    if stats_file:
+        stats = {}
+        if os.path.isfile(stats_file):
+            with open(stats_file, "r") as f:
+                stats = json.loads(f.read())
+        if stats_key:
+            stats[stats_key] = my_stats
+        else:
+            stats.update(my_stats)
+        open(stats_file, "w").write(json.dumps(stats))
+
+
+
+def cmp2dark(new_list="T2_US_Purdue_2021_06_18_02_28_D.list",
+             old_list="T2_US_Purdue_2021_06_17_02_28_D.list",
+             comm_list="out_D.list", stats_file="test_stats.json"):
+
+    t0 = time.time()
+    stats_key = "cmp2dark"
+    my_stats = stats = None
+    op = "and"
+
+    with open(new_list, "r") as a_list, open(old_list, "r") as b_list,\
+         open(comm_list, "w") as out_list:
+
+        if stats_file is not None:
+            stats = Stats(stats_file)
+            my_stats = {
+                "elapsed": None,
+                "start_time": t0,
+                "end_time": None,
+                "new_list": new_list,
+                "old_list": old_list,
+                "out_list": out_list.name,
+                "status": "started"
+            }
+            stats[stats_key] = my_stats
+
+        a_set = set(line.strip() for line in a_list)
+        b_set = set(line.strip() for line in b_list)
+
+
+# The intersection of the two sets is what can be deleted
+        out_set = a_set & b_set
+        out_list.writelines("\n".join(sorted(list(out_set))))
+
+    t1 = time.time()
+
+    if stats_file:
+        my_stats.update({
+            "elapsed": t1-t0,
+            "end_time": t1,
+            "status": "done"
+        })
+        stats[stats_key] = my_stats
+###############################
+### This was Igor's Stats class
+###############################
+
+
+####################
+### from Deckard
+####################
+def parse_filename(fn):
```

Comment:
Parse filename into what?
Maybe create a new class for the file identifiers. This way the returned information is bagged up into one variable instead of four when the function is called.

Reply:
The logic here is common with the Consistency Scanner, so it can be changed in the future in both places.
Adding a  TODO: tag for that purpose.